x-pack/plugins/endpoint/server/types.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { LoggerFactory } from 'kibana/server';
+import { EndpointConfigType } from './config';
+
+export interface EndpointAppContext {
+  logFactory: LoggerFactory;
+  config(): Promise<EndpointConfigType>;
+}
+
+export class EndpointAppConstants {
+  static ENDPOINT_INDEX_NAME = 'endpoint-agent*';
+}
+
+export interface EndpointData {
```

Comment:
a handful of these fields will likely be dropped for the first release

Reply:
we will probably just need something similar to `machine_id`, the items under `host`, perhaps `active_directory`, `policy`

We can remove some items in another PR, but we likely won't be supplying all of this to start

Reply:
Let talk and agree and we can remove in the next go around.